Update the documentation to match the current state of the mapping strategy (#356)
tweag/FawltyDeps: README.md

multiple-modules = ["module1", "module2"]
```

The provided mapping can be complete or partial. In the case where some of the
dependencies defined in your project configuration cannot be resolved,
FawltyDeps will attempt to resolve them using the sequence of resolvers 
illustrated in the diagram above.

Caution when using your mapping is advised: As illustrated in the diagram, the
user-defined mapping takes precedence over the other resolvers documented
above. For example, if the mapping file has some stale/incorrect mapping
entries, they will _not_ be resolved by the Python environment resolver (which
is usually more accurate).

Note that in the absence of a custom mapping, FawltyDeps will skip this resolver

> to the next one in the sequence, that is, mapping via local Python environments.

Author: Removed in [5b01a93](https://github.com/tweag/FawltyDeps/pull/356/commits/5b01a93398657401a7df59db5dbc424110086cb5).

#### Identity mapping

When unable to find an installed package that corresponds to a declared
dependency either via a user-provided mapping or local Python environments,
FawltyDeps will fall back to one of two strategies. "Identity mapping", which we
present in this section is the default fallback strategy. We discuss the other
strategy in the next subsection.

Identity mapping relies on the simplistic assumption that the dependency provides
a single import of the same name, i.e. it will expect that when you depend on
`some_package`, then that should correspond to `import some_package` statements
in your code.

This assumption is correct for many packages and it allows FawltyDeps to
produce results (albeit sometimes inaccurate ones) when the current Python
environment does not contain all of your declared dependencies.

To ensure correctness, however, refer to the next subsection outlining the other
fallback strategy.

> #### Mapping by temporarily installing packages
> Your local Python environements might not always have all your project's

Reviewer: Nitpicking to the extreme 😲 
```suggestion
#### Mapping by temporarily installing packages

Your local Python environements might not always have all your project's
```

dependencies installed. Assuming that you don’t want to go through the
bother of installing packages manually, and you also don't want to rely on
the inaccurate identity mapping as your fallback strategy, you can use the

> `--install-deps` option, introduced in v0.13.0. This will `pip install`

Author: This release does not exist yet so maybe I should drop this until it's out.

Reviewer: The README should avoid talking about version differences. Since this file is part of the codebase, it should simply reflect the current state (i.e. that `--install-deps` already exists):

```suggestion
`--install-deps` option. This will `pip install`
```

missing dependencies (from [PyPI](https://pypi.org/), by default) into a
_temporary virtualenv_, and allow FawltyDeps to use this to come up with the
correct mapping.

Since this is a potentially expensive strategy (e.g. downloading packages from
PyPI), we have chosen to hide it behind the `--install-deps` command-line
option. If you want to always enable this option, you can set the corresponding
`install_deps` configuration variable to `true` in the `[tool.fawltydeps]`
section of your `pyproject.toml`.

To customize how this auto-installation happens (e.g. use a different package index),
you can use [pip’s environment variables](https://pip.pypa.io/en/stable/topics/configuration/).

Note that we’re never guaranteed to be able to resolve _all_ dependencies with
this method: For example, there could be a typo in your `requirements.txt` that
means a dependency will _never_ be found on PyPI, or there could be other
circumstances (e.g. network issues or restrictions in your CI environment) that
prevent this strategy from working at all. 

> In this case, FawltyDeps will throw an error and will not produce an
> unused/undeclared report.

Reviewer: ```suggestion
In this case, FawltyDeps will throw an error and abort.
```
